src/Domain/Query/Query.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+declare(strict_types=1);
+
+namespace JeroenG\Explorer\Domain\Query;
+
+use JeroenG\Explorer\Domain\Syntax\Sort;
+use JeroenG\Explorer\Domain\Syntax\SyntaxInterface;
+
+class Query implements SyntaxInterface
+{
+    private ?int $offset = null;
+
+    private ?int $limit = null;
+
+    private array $fields = [];
+
+    /** @var Sort[] */
+    private array $sort = [];
+
+    private SyntaxInterface $query;
+
+    public static function with(SyntaxInterface $syntax)
```

Comment:
Should this be part of the interface since it is public?

Reply:
No this is a quick helper to construct a Query , not the interface.